### PR TITLE
Add: Fluidsynth music driver support (jmak)

### LINKED ISSUE #6611: Fluidsynth music driver
**_LordAro_ opened the ticket and wrote:**

> https://www.tt-forums.net/viewtopic.php?f=33&t=57037 by jmak
> 
> Last version attached. Will need updating

**Attachments**
- [0001-fluidsynth-support.patch](https://bugs.openttd.org/task/6611/getfile/10794/0001-fluidsynth-support.patch) (11.11 KiB)

**Reported version**: _trunk_
**Operating system**: _All_

---

### PR DESCRIPTION
**Pull request by LordAro**
Heavily based on the jmak's original patch. Only changed to update to @nielsmh's new music system, and a couple of comment tweaks

Build infrastructure might want altering if we want to actually test it/depend on it properly

Closes #6611

### CODE REVIEW

**Comment by nielsmh** [comment]
I'll find some time to test and review this one soon. :)

**Comment by nielsmh** [comment]
Cursory testing on Fedora is positive. Configure detects the library automatically (with `fluidsynth-devel` package installed), and it gets picked as default at runtime. Hasn't had any problems finding music in any of the usual locations, and handles both SMF and MPSMIDI sets.

File: src/music/fluidsynth.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	NULL
+};
+
+const char *MusicDriver_FluidSynth::Start(const char * const *param)
+{
+	const char *driver_name = GetDriverParam(param, "driver");
+	const char *sfont_name = GetDriverParam(param, "soundfont");
+	int sfont_id;
+
+	if (!driver_name) driver_name = "alsa";
```
Comment (nielsmh) [inline src/music/fluidsynth.cpp:51]: Should this perhaps be a configure option? Multiple reasons, first to let package maintainers pick an appropriate default for the system (many will want PulseAudio, I imagine), and second to better support other OS where ALSA does not exist, such as FreeBSD, Windows, and maybe macOS.

Comment (LordAro) [reply src/music/fluidsynth.cpp:51]: Possible, but given the hardcoded soundfont locations just above, it's already not going to be "easy" to use on systems other than Linux. Maybe TMWFTLB

Comment (nielsmh) [reply src/music/fluidsynth.cpp:51]: Right, leave it as-is, and if some package maintainer asks we can make a patch for this. The other alternative is taking the rendered music into buffers and mix internally in our own mixer, and share output that way. That's too much to add to this patch, but something maybe worth pursuing later.

**Review by nielsmh — APPROVED** [review]

**Pull request merged**